test/integration/framework/apiserver.go

```diff
@@ -65,6 +72,25 @@ func RunControlPlane(t *testing.T) (*rest.Config, StopFunc) {
 	if err != nil {
 		t.Fatalf("failed to start control plane: %v", err)
 	}
+
+	// set up test namespace
+	cl, err := client.New(config, client.Options{Scheme: api.Scheme})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, namespace := range testNamespaces {
+		err = cl.Create(context.Background(), &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		})
+
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
```

Comment:
I think it makes sense to keep this as part of the test itself, rather than shared across.

This makes it much harder to reason about where namespaces are coming from, and when they get created/are shared.

Each test should be self contained, create all it's resources it needs for that test, then clean up after itself.

Reply:
Agreed, will roll that back and add namespace creation where this wasn't done before

Reply:
Addressed in https://github.com/jetstack/cert-manager/pull/3166/commits/88dcb326f9265642813e4c62308fed6dbcc2c08a
sorry that it made the PR bigger, 1.19's framework handles namespaces correct ;)